e2e/framework/utils/k8s_helper.go

```diff
@@ -251,3 +251,60 @@ func (k8sh *K8sHelper) IsThirdPartyResourcePresent(tprname string) bool {
 
 	return false
 }
+
+func (k8sh *K8sHelper) GetPodDetails(podNamePattern string, namespace string) (string, error) {
+	cmdArgs := []string{"get", "pods", "-l", "app=" + podNamePattern, "-o", "wide"}
+	if namespace != "" {
+		cmdArgs = append(cmdArgs, []string{"-n", namespace}...)
+	}
+	sout, serr, status := ExecuteCmd("kubectl", cmdArgs)
+	if status != 0 {
+		return serr, fmt.Errorf("Cannot find service")
```

Comment:
error message refers to "service" but this function is for pod details.  Also consider including the `podNamePattern` and `namespace` in the error message.